Search, One-page forms and User mgmt section review (#10)
ezsystems/commerce-documentation: docs/guide/one_page_forms/one_page_forms.md

# One-page forms

eZ Commerce offers a unified way of handling all one-page forms.

One-page forms have several common criteria:

- They are displayed on a single page.
- After submitting some processes are executed in the backend.
- After server response the user sees a confirmation page with a success or error message.

> eZ Commerce uses the [Symfony forms](http://symfony.com/doc/current/forms.html) as a part of the solution,

Reviewer: ```suggestion
eZ Commerce uses the [Symfony forms](http://symfony.com/doc/3.4/forms.html) as a part of the solution,
```

so you can take advantage of what Symfony forms offer:

- easy handling with form entities
- simple and dynamic rendering with form types
- automatic form validation
- validation groups and much more

Additionally, you can pre-fill the form when it is loaded for the first time.
This gives you the opportunity to improve the user experience by offering default values.
The process that pre-fills the form is called `preDataProcessor`.

You can define processes that are executed after the form is submitted.
These processes are services that are executed is a sequence and are able to exchange the data.
These processes are called `dataProcessors`.

If process execution is not successful, the process can store an error message
and display it on the confirmation page.

You can decide what happens after the form is submitted - on server response.

> You can differ between valid and invalid responses and change the behavior according your needs.

Reviewer: ```suggestion
You can differ between valid and invalid responses and change the behavior according to your needs.
```

> Usually a confirmation page is displayed, but you can choose one of following behaviors:

Reviewer: ```suggestion
Usually a confirmation page is displayed, but you can choose one of the following behaviors:
```
